Marlin/temperature.h

```diff
@@ -220,6 +223,18 @@ class Temperature {
      */
     static void manage_heater();
 
+    // preheating functions
+    static bool is_preheating(int hotend) {
+      #ifdef MILLISECONDS_PREHEAT_TIME
+        return (preheatStartTime[hotend] > 0) && (millis() - preheatStartTime[hotend]) <= MILLISECONDS_PREHEAT_TIME;
+      #else
+        return false;
+      #endif
+    }
+
+    static void start_preheat_time(int hotend) { preheatStartTime[hotend] = millis(); }
+    static void reset_preheat_time(int hotend) { preheatStartTime[hotend] = 0; }
```

Comment:
We prefer ending-times, as they require less computation per loop.
And we have some macros for this. So this would be better:
```cpp
// preheating functions
static bool is_preheating(int hotend) {
  #ifdef MILLISECONDS_PREHEAT_TIME
    return (preheatEndTime[hotend] && PENDING(millis(), preheatEndTime[hotend]));
  #else
    return false;
  #endif
}

static void start_preheat_time(uint8_t hotend) { preheatEndTime[hotend] = millis() + MILLISECONDS_PREHEAT_TIME; }
static void reset_preheat_time(uint8_t hotend) { preheatEndTime[hotend] = 0; }
```